### PR TITLE
improve(bridges): add special USDC case for the ScrollBridge.

### PR DESCRIPTION
**Pull request by bmzig**
This is a simple fix to the issue we had for resolving a USDC address for scroll, whereby we would include a special case in `resolveL2TokenAddress` in BaseBridgeAdapter. This moves this logic to the Scroll bridge so this conditional won't be triggered by all bridges.

### CODE REVIEW

**Review by james-a-morris — COMMENTED** [review]
Is this change a refactor?

**Comment by bmzig** [comment]
> Is this change a refactor?

Essentially. It's a fix for the todo brought up in this PR https://github.com/across-protocol/relayer/pull/1708

File: src/adapter/bridges/ScrollERC20Bridge.ts [hunk 1]
```diff
@@ -127,4 +129,11 @@ export class ScrollERC20Bridge extends BaseBridgeAdapter {
   protected getScrollGatewayRouter(): Contract {
     return this.scrollGatewayRouter;
   }
+
+  protected override resolveL2TokenAddress(l1Token: string): string {
+    if (compareAddressesSimple(TOKEN_SYMBOLS_MAP.USDC.addresses[this.hubChainId], l1Token)) {
+      return TOKEN_SYMBOLS_MAP.USDC.addresses[this.l2chainId]; // Scroll only has one USDC token type.
+    }
+    return super.resolveL2TokenAddress(l1Token);
```
Comment (james-a-morris) [inline src/adapter/bridges/ScrollERC20Bridge.ts:137]: Good call. Thoughts on
```suggestion
  protected override resolveL2TokenAddress(l1Token: string): string {
  
    if () {
      return TOKEN_SYMBOLS_MAP.USDC.addresses[this.l2chainId]; // Scroll only has one USDC token type.
    }
    return super.resolveL2TokenAddress(l1Token);
```

**Review by james-a-morris — APPROVED** [review]

**Review by pxrl — APPROVED** [review]
Nice and clean!

**Pull request merged**